src/core_plugins/kibana/ui_setting_defaults.js

```diff
@@ -114,6 +114,15 @@ export function getUiSettingDefaults() {
         'When set to true, filter(s) will be ignored for a visualization ' +
         'when the visualization\'s index does not contain the filtering field.'
     },
+    'courier:setRequestPreference': {
+      value: true,
+      description: '<a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-preference.html" target="_blank" rel="noopener noreferrer">Request Preference</a> ' +
+        ' controls the shard copies used for search execution.' +
+        ' Set to true to execute all search requests on the same shards. ' +
```

Comment:
I think the description should mention the original reason we started using `preference`, which I believe was caching (Please correct me if I'm wrong @rashidkpc).

Like your description states, not sending `preference` sort of randomizes the node that handles the request, which means that the caches from a previous request might not be reusable on a second request and will need to be rebuilt.

Reply:
What about this 

"Set to true to execute all search requests on the same shards. This has the benefit of reusing Shard Request Cache across requests. Set to false to have search request execution randomized among all available shard copies. Setting to false may provide better performance since requests can be spread across all shard copies but may result in inconsistent results as different shards may be in different refresh states."

Reply:
I wonder how many requests can actually even use the cache since... "Most queries that use now (see Date Mathedit) cannot be cached." [from warning in shard-request-cache.html](https://www.elastic.co/guide/en/elasticsearch/reference/current/shard-request-cache.html)

Reply:
There are other caches, like the caches from filters, which I'm almost positive are not impacted by using `now`.

Reply:
👍